Refactoring reporting ID capability in patterned mesh generators to support multiple ID assignment (#26859)
idaholab/moose: modules/reactor/doc/content/source/meshgenerators/PatternedCartesianMeshGenerator.md

For example, the reporting IDs for individual pins (`pin_id`) can be assigned when assemblies are built because the IDs for pin level are uniquely determined from the pin arrangement within each assembly type.
Similarly, the assembly reporting IDs (`assembly_id`) are assigned in the core construction process.

> The multiple `reporting IDs` can be assigned by defining the multiple name of the reporting ID variable, which are provided through the[!param](/Mesh/PatternedCartesianMeshGenerator/id_name).

Reviewer: ```suggestion
The multiple `reporting IDs` can be assigned by defining the multiple names of the reporting ID variable, which are provided through the[!param](/Mesh/PatternedCartesianMeshGenerator/id_name).
```
it s not a variable though. do you mean parameter here or EEID?

> The correponding assignment scheme should be provided in [!param](/Mesh/PatternedCartesianMeshGenerator/assign_type) for each reporting ID names, accordingly.

Reviewer: ```suggestion
The corresponding assignment scheme should be provided in [!param](/Mesh/PatternedCartesianMeshGenerator/assign_type) for each reporting ID names, accordingly.
```

> In case of multiple `manual` assign types are used, the same number of manual ID patterns should be provided in [!param](/Mesh/PatternedCartesianMeshGenerator/id_pattern).

Reviewer: ```suggestion
In the case that multiple `manual` `assign_type`s are used, the same number of manual ID patterns should be provided in [!param](/Mesh/PatternedCartesianMeshGenerator/id_pattern).
```

Each manual pattern in [!param](/Mesh/PatternedCartesianMeshGenerator/id_pattern) should be separated by using the delimiter '|'.
These defined ID patterns are sequentially assigned to the reporting IDs having `manual` assignment scheme.
The below is an example of using multiple `reporting ID` assignment.

> Here, `manual_1_id` uses the first pattern in defined in [!param](/Mesh/PatternedCartesianMeshGenerator/id_pattern), and `manaul_2_id` uses the second one.

Reviewer: ```suggestion
Here, `manual_1_id` uses the first pattern in defined in [!param](/Mesh/PatternedCartesianMeshGenerator/id_pattern), and `manual_2_id` uses the second one.
```

!listing!
id_name = 'manual_1_id cell_id manaul_2_id'
assign_type 'manual cell manual'
id_pattern = '1 2;
              3 4|
              4 3;
              2 1;
!listing-end!

> 

Reviewer: apply comments to the next file
